lib/good_job/railtie.rb

```diff
@@ -23,6 +23,11 @@ class Railtie < ::Rails::Railtie
     end
 
     config.after_initialize do
+      GoodJob.logger = Rails.application.config.good_job.logger if Rails.application.config.good_job.logger.present?
+      GoodJob.on_thread_error = Rails.application.config.good_job.on_thread_error if Rails.application.config.good_job.on_thread_error.present?
+      GoodJob.preserve_job_records = Rails.application.config.good_job.preserve_job_records if Rails.application.config.good_job.preserve_job_records.present?
+      GoodJob.retry_on_unhandled_error = Rails.application.config.good_job.retry_on_unhandled_error if Rails.application.config.good_job.retry_on_unhandled_error.present?
```

Comment:
This is the meat of the change, with the effect of overriding the values assigned to `GoodJob` with values in Rails config.